lib/plugins/acts_as_activity_provider/lib/acts_as_activity_provider.rb

```diff
@@ -191,13 +191,13 @@ def restrict_projects_by_user(options, user, query)
               allowed_projects = []
 
               user.projects_by_role.each do |role, projects|
-                allowed_projects << projects.map(&:id) if role.allowed_to?(perm.name)
+                allowed_projects << projects.map(&:id) if perm && role.allowed_to?(perm.name)
```

Comment:
If we do it like this then stmt in line 197 will be `projects.id in ()` which would forbid all projects. Is that how it should be if the permission is missing? Or should one rather get access to the projects?